local_flow/rec/src/rec_baseline_flow.py

```diff
@@ -129,40 +129,40 @@ def get_model_config(self):
     # wrap batch in a switch to allow easy local testing
     @enable_decorator(batch(gpu=1, memory=60000, cpu=8, image=os.getenv('BASE_IMAGE')), flag=os.getenv('EN_BATCH'))
     # @ environment decorator used to pass environment variables to Batch instance
-    @environment(vars={'WANDB_API_KEY': os.getenv('WANDB_API_KEY'),
+    @environment(vars={'NEPTUNE_PROJECT': os.getenv('NEPTUNE_PROJECT'),
+                       'NEPTUNE_API_TOKEN': os.getenv('NEPTUNE_API_TOKEN'),
+                       'NEPTUNE_CUSTOM_RUN_ID': os.getenv('NEPTUNE_CUSTOM_RUN_ID'),
+                       'WANDB_API_KEY': os.getenv('WANDB_API_KEY'),
                        'WANDB_ENTITY': os.getenv('WANDB_ENTITY'),
                        'BASE_IMAGE': os.getenv('BASE_IMAGE'),
                        'EN_BATCH': os.getenv('EN_BATCH')})
     # un-comment if provided image does not contain required packages
-    # @pip(libraries={'wandb': '0.10.30', 'gensim': '4.0.1'})
+    # @pip(libraries={'wandb': '0.10.30', 'gensim': '4.0.1', "neptune-client": "0.13.3", "neptune-tensorflow-keras": "0.9.9"})
     @step
     def train_model(self):
         """
         Train a Prod2Vec or ProdB model.
         """
-        import os
-        import wandb
+        from utils import ExperimentTracker
         from model import train_prod2vec_model, train_prodb_model
 
-        assert os.getenv('WANDB_API_KEY')
-        assert os.getenv('WANDB_ENTITY')
-
-        # initialize wandb for tracking
-        wandb.init(entity=os.getenv('WANDB_ENTITY'),
-                   project="recommendation-{}".format(self.model_choice),
-                   id=current.run_id,
-                   config=self.config,
-                   resume='allow',
-                   reinit=True)
-
+        # Initialize neptune or wandb for tracking
+        tracker = ExperimentTracker(
+            current_run_id=current.run_id,
```

Comment:
One last concern about this. If we take the decision of which Experiment tracker to use inside `ExperimentTracker` then we will lose the explicitness. The developer will have to go inside the `ExperimentTracker` in order to find the decision logics. I think it would be better to move the decision logics of which experiment tracker to use outside: at the level of the flow that way we are fully explicit about what gets passed into it. 

Remembering python's golden rules: "explicit is better than implicit"

Reply:
Done ✅